tutorials/e2e-huggingface-summarization/src/run_summarization/run.py

```diff
@@ -0,0 +1,377 @@
+        default=None, metadata={"help": "config of input HF dataset"}
+    )
+    text_column: Optional[str] = field(
+        default=None, metadata={"help": "the key for text column"}
+    )
+    summary_column: Optional[str] = field(
+        default=None, metadata={"help": "the key for summary column"}
+    )
+    source_prefix: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "A prefix to add before every source text (useful for T5 models)."
+        },
+    )
+    preprocessed_datasets: Optional[str] = field(
+        default=None, metadata={"help": "path to preprocesed datasets"}
+    )
+    train_file: Optional[str] = field(
+        default=None, metadata={"help": "path to train file(json)"}
+    )
+    validation_file: Optional[str] = field(
+        default=None, metadata={"help": "path to validation file(jsonl)"}
+    )
+    test_file: Optional[str] = field(
+        default=None, metadata={"help": "path to test file(jsonl)"}
+    )
+    max_input_length: Optional[int] = field(
+        default=1024,
+        metadata={"help": "max input sequence length after tokenization."},
+    )
+    max_target_length: Optional[int] = field(
+        default=128,
+        metadata={"help": "maxi sequence length for target text after tokenization."},
+    )
+    max_samples: Optional[int] = field(
+        default=None,
+        metadata={"help": "limit the number of samples for faster run."},
+    )
+    num_beams: Optional[int] = field(
+        default=None,
+        metadata={"help": "limit the number of samples for faster run."},
+    )
+    padding: Optional[str] = field(
+        default="max_length", metadata={"help": "padding setting"}
+    )
+
+    def __post_init__(self):
+        if (
+            self.dataset_name is None
+            and self.preprocessed_datasets is None
+            and self.train_file is None
+            and self.validation_file is None
+        ):
+            raise ValueError(
+                "Need either a dataset name, preprocessed_datasets, or a training/validation file."
+            )
+
+
+@dataclass
+class ModelArgs:
+    trained_model_path: str
+    registered_model_name: Optional[str] = field(
+        default=None, metadata={"help": "registered model name"}
+    )
+    model_name: Optional[str] = field(default=None, metadata={"help": "model name"})
+    model_path: Optional[str] = field(
+        default=None, metadata={"help": "path to model file"}
+    )
+
+
+run = Run.get_context()
+logger = logging.getLogger(__name__)
+nltk.download("punkt")
+
+
+def print_gpu_utilization():
+    nvmlInit()
+    handle = nvmlDeviceGetHandleByIndex(0)
+    info = nvmlDeviceGetMemoryInfo(handle)
+    print(f"GPU memory occupied: {info.used//1024**2} MB.")
+
+
+def print_summary(result):
+    print(f"Time: {result.metrics['train_runtime']:.2f}")
+    print(f"Samples/second: {result.metrics['train_samples_per_second']:.2f}")
+    print_gpu_utilization()
+
+
+def main():
```

Comment:
I feel this function is a little long, I'd rather have clear sections in the code when browsing it as a reference code. But we can also do that in the mlops repo and project back code here once it's structured.